vertx-web/src/main/java/io/vertx/ext/web/handler/sockjs/impl/BaseTransport.java

```diff
@@ -155,20 +154,22 @@ static void setJSESSIONID(SockJSHandlerOptions options, RoutingContext rc) {
   }
 
   static void setCORS(RoutingContext rc) {
-    HttpServerRequest req = rc.request();
-    String origin = req.headers().get("origin");
-    if (origin == null) {
-      origin = "*";
-    }
-    Utils.addToMapIfAbsent(req.response().headers(), "Access-Control-Allow-Origin", origin);
-    if ("*".equals(origin)) {
-      Utils.addToMapIfAbsent(req.response().headers(), "Access-Control-Allow-Credentials", "false");
-    } else {
-      Utils.addToMapIfAbsent(req.response().headers(), "Access-Control-Allow-Credentials", "true");
-    }
-    String hdr = req.headers().get("Access-Control-Request-Headers");
-    if (hdr != null) {
-      Utils.addToMapIfAbsent(req.response().headers(), "Access-Control-Allow-Headers", hdr);
+    if (rc.get("corsHeadersWritten") == null || !((boolean)rc.get("corsHeadersWritten"))) {
```

Comment:
I'd suggest to make `corsHeadersWritten` a `public static final String` constant so there's no future potential bugs due to typpo's.